src/Quantum/weight_reduction.jl

```diff
@@ -620,208 +620,190 @@ function _cycle_basis_decongestion(_edges::Vector{Tuple{T, T}}) where T
-thickening and choosing heights.
+Return the result of coning on `H_X` and `H_Z` by reducing the `Z` stabilizers in `row_indices`. The
+optional argument `rng` can be used to make the output of this function reproducible.
 """
-function coning(H_X::T, H_Z::T, row_indices::AbstractVector{Int}; l::Int = 0, target_q_X::Int = 3) where T <: CTMatrixTypes
+function coning(H_X::T, H_Z::T, row_indices::AbstractVector{Int}; rng::AbstractRNG = Random.seed!()) where T <: CTMatrixTypes
     
     F = base_ring(H_X)
     n_X, n = size(H_X)
     n_Z = size(H_Z, 1)
 
     issubset(row_indices, 1:n_Z) || throw(DomainError(row_indices, "Choice of Z stabilizers is out of bounds."))
-    target_q_X >= 3 || throw(DomainError(target_q_X, "Can't target a q_X below 3."))
-
-    # Create the spaces (B_i)_1. Each space is the support of a Z stabilizer
-    Bi1 = [getindex.(findall(!iszero, H_Z[i, :]), 2) for i in row_indices]
-
-    # Create the spaces (B_i)_0
-    # TODO: This assumes overlap of Bi1[i] and any X stabilizer is 0 or 2 qubits. This is true in Hastings algorithm but won't be true in general. Need to extend to work in the general case.
-    Xsupports = [getindex.(findall(!iszero, H_X[i, :]), 2) for i in 1:n_X]
-    all(all(length(X ∩ Q) < 3 for X in Xsupports) for Q in Bi1) || throw(DomainError(row_indices, "Z stabilizers chosen for reducing must have overlap of at most 2 with all X stabilizers."))
-    Bi0 = [Tuple{Int, Tuple{Int, Int}}[(i, (sort(X ∩ Q)...,)) for (i, X) in enumerate(Xsupports) if !iszero(length(X ∩ Q))] for Q in Bi1]
-
-    # Create the spaces (B_i)_{-1} and append some edges to (B_i)_0
-    Bim1 = [Vector{Int}[] for _ in eachindex(Bi0)]
-    for i in eachindex(Bim1)
-        # extract edges from Bi0[i] and find a cycle basis c for the graph they define
-        edges = getindex.(Bi0[i], 2)
-        isempty(edges) && (edges = Tuple{Int, Int}[];)
-        unique_edges = unique(edges)
 
-        # c = Grphs.cycle_basis(Grphs.SimpleGraph(Grphs.SimpleEdge.(unique_edges)))
-        c = _cycle_basis_decongestion(unique_edges)
-
-        # Bim1[i] = c
-
-        # cellulate big cycles
-        for j in eachindex(c)
-            if length(c[j]) > 4 # split it up
-                a = 1
-                b = length(c[j])
-                while b - a - 1 >= 2 # at least 2 points between a and b
-                    push!(Bim1[i], c[j][[a, a + 1, b - 1, b]])
-                    a += 1
-                    b -= 1
-                    if b - a > 1 # keep track of new edges that need added
-                        new_edge = (-1, (sort(c[j][[a, b]])...,))
-                        new_edge ∈ Bi0[i] || push!(Bi0[i], new_edge)
-                    end
-                end
-                if b - a - 1 > 0 # if there are any points left between a and b
-                    push!(Bim1[i], c[j][[a; a + 1:b - 1; b]])
-                end
-            else # keep the cycle as is
-                push!(Bim1[i], c[j])
-            end
+    hx = Int.(data.(H_X))
```

Comment:
data. is no longer compatible with the new version of Oscar. Let's use the provided _Flint...() methods

Reply:
latest commit brings in the _Flint... methods from the dev branch and uses them.